internal/injector/aspect/join/function.go

```diff
@@ -347,6 +357,73 @@ func (s *functionBody) Hash(h *fingerprint.Hasher) error {
 	return h.Named("function-body", s.Function)
 }
 
+// returnImplements matches functions where at least one return value's type
+// implements the specified interface.
+type returnImplements struct {
+	InterfaceName string
+}
+
+// ReturnImplements creates a FunctionOption that matches functions where at least one
+// return value implements the named interface.
+func ReturnImplements(interfaceName string) FunctionOption {
+	return &returnImplements{InterfaceName: interfaceName}
+}
+
+func (fo *returnImplements) impliesImported() []string {
+	pkgPath, _ := typed.SplitPackageAndName(fo.InterfaceName)
+	if pkgPath != "" {
+		return []string{pkgPath}
+	}
+	return nil
+}
+
+func (_ *returnImplements) packageMayMatch(_ *may.PackageContext) may.MatchType {
+	// Cannot reliably determine possibility of match based on package imports
+	// due to structural typing. A type can implement an interface without
+	// importing the interface's package.
+	return may.Unknown
+}
+
+func (_ *returnImplements) fileMayMatch(_ *may.FileContext) may.MatchType {
+	// Cannot reliably determine possibility of match based on file contents
+	// due to structural typing and type aliases.
+	return may.Unknown
+}
+
+func (fo *returnImplements) evaluate(info functionInformation) bool {
+	if info.Type.Results == nil || len(info.Type.Results.List) == 0 {
+		// No return values, no match.
+		return false
+	}
+
+	// Ensure the type resolver is available.
+	if info.typeResolver == nil {
+		return false
+	}
+
+	// Resolve the target interface name (e.g., "io.Reader", "error") to a types.Interface.
+	targetInterface, err := typed.ResolveInterfaceTypeByName(fo.InterfaceName)
+	if err != nil {
+		// If the interface name is invalid or cannot be resolved, we cannot match.
+		return false
+	}
+
+	for _, field := range info.Type.Results.List {
+		// For each return type (dst.Expr), resolve it to types.Type using the provided resolver
+		// and check if it implements the target interface.
+		if typed.ExprImplements(info.typeResolver, field.Type, targetInterface) {
+			return true // Found at least one implementing type, match!
+		}
+	}
+
+	// No return type matched.
+	return false
+}
+
+func (fo *returnImplements) Hash(h *fingerprint.Hasher) error {
+	return h.Named("return-implements", fingerprint.String(fo.InterfaceName))
```

Comment:
We are calling these _results_ not _return values_, because this is the language used in the Go language spec... So i'd call this `result-implements`, not `return-implements`...